trio/_subprocess.py

```diff
@@ -687,23 +719,28 @@ async def my_deliver_cancel(process):
             assert os.name == "posix"
             deliver_cancel = _posix_deliver_cancel
 
-    stdout_chunks = []
-    stderr_chunks = []
+    stdout_chunks: list[bytes | bytearray] = []
+    stderr_chunks: list[bytes | bytearray] = []
 
-    async def feed_input(stream):
+    async def feed_input(stream: SendStream) -> None:
         async with stream:
             try:
+                assert input is not None
                 await stream.send_all(input)
             except trio.BrokenResourceError:
                 pass
 
-    async def read_output(stream, chunks):
+    async def read_output(
+        stream: ReceiveStream,
+        chunks: list[bytes | bytearray],
+    ) -> None:
         async with stream:
             async for chunk in stream:
                 chunks.append(chunk)
 
     async with trio.open_nursery() as nursery:
-        proc = await open_process(command, **options)
+        # options needs a complex TypedDict.
+        proc = await open_process(command, **options)  # type: ignore
```

Comment:
restrict the `type: ignore`. Actually, we could maybe enable a setting that warns about unrestricted `type: ignore`s

Reply:
Unfortunately, since Unix has an overloaded `open_process` and Windows does not the type code varies depending on platform. Realised I can ignore the unused ignores though :P

Reply:
Fwiw you can ignore multiple type codes, just use a comma

Reply:
Yep, the issue is that on Windows you get `arg-type`, but on Linux/Mac you get `arg-type` + `call-overload`. So I had to use `type: ignore[arg-type, call-overload, unused-ignore]` to ignore the ignore not being used.